openedx/xblock-lti-consumer
# [BB-5559] Decouple LTI 1.3 from LTI Consumer XBlock functionality

### Description

The goal of the PR is to decouple the LTI 1.3 functionality from the XBlock and move the functionality to the Django Plugin, while keeping backward compatibility with the XBlock Handlers. The main consideration is that, this does not break any existing integrations. So all the endpoints and XBlock handlers are made backward compatible.

#### Changes introduced by the refactor

1. The functionality of LTI 1.3 Launch Handler is moved from the XBlock to the Django plugin
2. The functionality of the Access Token endpoint is moved from XBlock to the Django plugin
3. A [new URL format](https://github.com/openedx/xblock-lti-consumer/pull/254/files#diff-80779cdd3ff9ef517c54ff7142a76970028a4eaf8b155c013e2a4a3d54b015d0R40) using the `LtiConfiguration` ID is introduced for the Access Token endpoint and is used when a LTI Consumer is configured without a `location` allowing LTI integrations to be created without the XBlock context.
5. A [new URL format](https://github.com/openedx/xblock-lti-consumer/pull/254/files#diff-80779cdd3ff9ef517c54ff7142a76970028a4eaf8b155c013e2a4a3d54b015d0R24) using the `LtiConfiguration` ID is introduced for the Keyset Endpoint and is used with the `location` of the XBlock is not available in the configuration.

### Testing instructions

1. Setup an LTI Consumer block (using test tools like [Saltire](https://saltire.lti.app/) or [IMS Global Reference Tool](https://lti-ri.imsglobal.org/))
2. Verify the LTI launches are working as expected.
3. Install the PR branch in the test system `pip install git+https://github.com/open-craft/xblock-lti-consumer.git@tecoholic/BB-5559-lti1p3-refactor`. If you are using devstack, install it in both lms and studio containers.
4. Restart the services as necessary.
5. Ensure that all the LTI Blocks are working as before.
